AK/IpAddressCidr.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024, famfo <famfo@famfo.xyz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/IPv4Address.h>
+#include <AK/IPv6Address.h>
+#include <AK/Result.h>
+
+namespace AK {
+
+class [[gnu::packed]] IPv4AddressCidr;
+class [[gnu::packed]] IPv6AddressCidr;
+template<typename Address>
+class [[gnu::packed]] AddressTraits;
+
+template<OneOf<IPv4AddressCidr, IPv6AddressCidr> AddressFamily>
+class [[gnu::packed]] IPAddressCidr {
```

Comment:
This can go into a Details {} namespace since I think noone wants to access it directly.